commcare-cloud/commcare_cloud.py

```diff
@@ -0,0 +1,125 @@
+
+        def run_apply():
+            return ansible_playbook(args.environment, args.playbook, get_ansible_vault_password(), *unknown_args)
+
+        def get_ansible_vault_password():
+            if not get_ansible_vault_password.value:
+                get_ansible_vault_password.value = getpass.getpass("Vault Password: ")
+            return get_ansible_vault_password.value
+        get_ansible_vault_password.value = None
+
+        exit_code = 0
+
+        if args.skip_check:
+            user_wants_to_apply = ask('Do you want to apply without running the check first?')
+        else:
+            exit_code = run_check()
+            if exit_code == 1:
+                # this means there was an error before ansible was able to start running
+                exit(exit_code)
+                return  # for IDE
+            elif exit_code == 0:
+                puts(colored.green(u"✓ Check completed with status code {}".format(exit_code)))
+                user_wants_to_apply = ask('Do you want to apply these changes?')
+            else:
+                puts(colored.red(u"✗ Check failed with status code {}".format(exit_code)))
+                user_wants_to_apply = ask('Do you want to try to apply these changes anyway?')
+
+        if user_wants_to_apply:
+            exit_code = run_apply()
+            if exit_code == 0:
+                puts(colored.green(u"✓ Apply completed with status code {}".format(exit_code)))
+            else:
+                puts(colored.red(u"✗ Apply failed with status code {}".format(exit_code)))
+
+        exit(exit_code)
+
+
+class UpdateConfig(object):
+    @staticmethod
+    def make_parser(parser):
+        parser.add_argument('--skip-check', action='store_true', default=False)
+        parser.add_argument('--branch', default='master')
+
+    @staticmethod
+    def run(args, unknown_args):
+        args.playbook = 'deploy_localsettings.yml'
+        unknown_args += ('--tags=localsettings',)
+        AnsiblePlaybook.run(args, unknown_args)
+
+
+def git_branch():
+    return subprocess.check_output("git branch | grep '^*' | cut -d' ' -f2", shell=True,
+                                   cwd=os.path.expanduser('~/.commcare-cloud/ansible')).strip()
+
+
+def check_branch(args):
+    branch = git_branch()
+    if args.branch != branch:
+        if branch != 'master':
+            puts(colored.red("You are not on branch master. To deploy anyway, use --branch={}".format(branch)))
+        else:
+            puts(colored.red("You are on branch master. To deploy, remove --branch={}".format(branch)))
+        exit(-1)
+
+
+def main():
+    parser = ArgumentParser()
+    parser.add_argument('environment')
+    subparsers = parser.add_subparsers(dest='command')
+
+    AnsiblePlaybook.make_parser(subparsers.add_parser('ansible-playbook'))
+    UpdateConfig.make_parser(subparsers.add_parser('update-config'))
```

Comment:
What do you think of renaming "update-config" to "deploy-localsettings"? It's not obvious what "config" it's referring to without reading the code in this file.

Reply:
Oh, I see you mentioned this in the PR description. I'm not a fan of renaming for consistency with a best practice we're not following: [Store config in the environment](https://12factor.net/config) (aka _environment variables_). Also from that page "Note that this definition of “config” does not include internal application config, such as `config/routes.rb` in Rails"—I think that applies to django localsettings.

Reply:
So, I definitely think there's an argument to be made that whether it's industry practice or not, we call these things "localsettings" here at Dimagi, so we shouldn't confuse people by now calling them "config". That said, in my reading of that page, our `localsettings.py` files are absolutely "config"; we just aren't following their recommendation that config be stored in the environment. The negative example they gave of `config/routes.rb` would for us by `urls.py` files—i.e. "configuration" of the framework (rails/django), but an immutable part of your codebase. I'm currently still a fan of my choice not to call this deploy-localsettings (I think it keeps us focussed on the implementation rather than the intention; already one of the "localsettings" files update is formplayer's `applications.properties`). I guess it's inconvenient that I already rolled this out and announced it, but let me know if you feel strongly about pushing back on this name.

Reply:
I see your point about `routes.rb` being more like `urls.py`. I still think it's unnecessary to rename it since it requires more context to understand what it's actually doing, but it might just be opinion/personal preference.